wgpu-native/src/id.rs

```diff
@@ -0,0 +1,530 @@
+use crate::{Epoch, Id, Index, TypedId};
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+#[cfg(not(feature = "gfx-backend-gl"))]
+#[repr(transparent)]
+#[derive(Clone, Copy, Debug, Hash, PartialEq)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct InstanceId(Id);
```

Comment:
Can we do something that doesn't use macros but still avoids repeating all those things? I.e.
```rust
#[repr(transparent)]
#[derive(Clone, Copy, Debug, Hash, PartialEq)]
#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
pub struct GenericId<T>(Id);

pub type InstanceId = GenericId<InstanceHandle>;
pub type DeviceId = GenericId<DeviceHandle>;
...
```